src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -477,8 +482,11 @@ private bool ProcessBlock(Block block, ChainedHeader header)
 
                 int purgeRewindDataThreshold = Math.Min(this.consensusManager.Tip.Height - this.compactionTriggerDistance, this.lastSavedHeight);
 
-                if (purgeRewindDataThreshold % 1000 == 0)
+                if ((DateTime.Now - this.lastPurgeTime).TotalSeconds > PurgeIntervalSeconds)
```

Comment:
I've just seen that this file uses the default DateTimeProvider everywhere, so this makes sense, but it might be helpful for us in the future if this uses an injected IDateTimeProvider to avoid unpredictable tests

Reply:
How could using default DateTimeProvider bring unpredictability? 
Maybe I'm not seeing the possibility, could you please provide an example? 

Also even if during tests we get purging at random times it shouldn't affect the test overall. 
In some tests we setup the time provider so if we use it here we might occasionally end up triggering purging on every new block